### DISCUSSION TITLE
Form() `alias` ignored in new fastapi 0.113.0

### DISCUSSION THREAD

**Discussion by fnep** [fastapi/fastapi#12139] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Form
from typing import Annotated

from fastapi import FastAPI

app = FastAPI()


@app.post("/")
def read_root(
        form_session_id: Annotated[str | None, Form(alias="sessionid")] = None,
):
    return form_session_id
```

### Description

Sending a request with `sessionid` in the form data to this root endpoint returns null:

```bash
> curl  --form sessionid=123qwe http://127.0.0.1:8000/
null
```

With `fastapi==0.112.2` it returns `123qwe`.

Interestingly with `0.113.0` and a "blind" declaration of sessionid as form value it also works.

```python3
@app.post("/")
def read_root(
        form_session_id: Annotated[str | None, Form(alias="sessionid")] = None,
        sessionid: Annotated[str | None, Form()] = None,
):
    return form_session_id
```

(note that im not returning the blind sessionid but the form_session_id, anyhow this makes the difference).

### Operating System

Linux, macOS

### FastAPI Version

0.113.0

### Pydantic Version

2.9.0

### Python Version

3.12.1

**Comment by sinisaos** [comment, marked as answer]
@fnep Yes, you are right. I think this regression problem lies [in this line](https://github.com/fastapi/fastapi/blob/d86f6603029def91e0798ca42f5fd12eff13c87b/fastapi/dependencies/utils.py#L791). I think this should be  

```python
if value is not None:
    values[field.alias] = value
```
because `alias` in `first_field_info` is set to alias if we specified in `Form(alias=...)`, and if not, field name is set to alias in `first_field_info`. Can you patch a local fastapi install with that change and give it a try?

**Reply by fnep** [reply]
Yes, indeed this fixes it for me.
Are you going to make a PR for this?

**Reply by tonnico** [reply]
I can confirm, that this would fix my issues also.

**Reply by tiangolo** [reply]
This was fixed in https://github.com/fastapi/fastapi/pull/12194 by @Wurstnase, thanks! :bow: 

It will be available in FastAPI 0.114.2 released in the next few hours. :rocket:

**Comment by sinisaos** [comment]
@fnep I see a few complaints about the regression in version `0.113.0` and probably the author will fix it all so there is no need for my PR. @tiangolo could you please consider making changes from this discussion in a future release. Thanks in advance.
